src/hotspot/share/runtime/globals.hpp

```diff
@@ -554,6 +554,12 @@ const intx ObjectAlignmentInBytes = 8;
           "directory) of the dump file (defaults to java_pid<pid>.hprof "   \
           "in the working directory)")                                      \
                                                                             \
+  product(intx, HeapDumpGzipLevel, 0, MANAGEABLE,                           \
+          "When HeapDumpOnOutOfMemoryError is on, the gzip compression "    \
+          "level of the dump file. 0 (the default) disables gzip "          \
+          "compression. Otherwise the level must be between 1 and 10.")     \
+          range(0, 10)                                                      \
```

Comment:
Isn't the range 0-9? Although `HeapDumper::dump()` imposes no upper limit (it seems it should), the `HeapDumpDCMD` sets the upper limit to 9. See `HeapDumpDCmd::execute()`.

Reply:
You're right. 9 is the maximum compression level supported by zlib. I will update the PR.